packages/jupyter-ai/jupyter_ai/models.py

```diff
@@ -263,3 +263,45 @@ class ListSlashCommandsEntry(BaseModel):
 
 class ListSlashCommandsResponse(BaseModel):
     slash_commands: List[ListSlashCommandsEntry] = []
+
+
+class ListOptionsEntry(BaseModel):
+    id: str
+    # includes the command prefix. e.g. "/clear", "@file".
+    label: str
+    description: str
+    only_start: bool
+    # only allows autocomplete to be triggered if command is at start of input
+
+    @classmethod
+    def from_command(
+        cls,
+        id: str,
+        description: str,
+        only_start: bool = False,
+        requires_arg: bool = False,
+    ):
+        label = id + (":" if requires_arg else " ")
+        return cls(id=id, description=description, label=label, only_start=only_start)
+
+    @classmethod
+    def from_arg(
+        cls,
+        id: str,
+        description: str,
+        arg: str,
+        only_start: bool = False,
+        is_complete: bool = True,
+    ):
+        arg = arg.replace("\\ ", " ").replace(" ", "\\ ")  # escape spaces
+        label = id + ":" + arg + (" " if is_complete else "")
+        return cls(
+            id=id,
+            description=description,
+            label=label,
+            only_start=only_start,
+        )
+
```

Comment:
Is it possible to define `arg` and `label` as computed properties instead of requiring class methods to create an instance of `ListOptionsEntry`? I'm actually not sure if this is possible in Pydantic v1, so feel free to push back if you find out it's not possible with the v1 API.

Reply:
I'll instead move this logic out of this class and implement them where they are used. Originally they were used in different files so I needed to central place to store them.